lib/Backend/NativeCodeGenerator.cpp

```diff
@@ -2802,25 +2810,40 @@ void
 NativeCodeGenerator::ProfilePrint()
 {
     Js::ScriptContextProfiler *codegenProfiler = this->backgroundCodeGenProfiler;
+
     if (Js::Configuration::Global.flags.Verbose)
     {
         //Print individual CodegenProfiler information in verbose mode
         while (codegenProfiler)
         {
-            codegenProfiler->ProfilePrint(Js::Configuration::Global.flags.Profile.GetFirstPhase());
+            if (codegenProfiler->IsInitialized())
```

Comment:
You are right, this check is redundant. No merge action here so the check in ScriptContextProfiler::ProfilePrint is enough.

---
In reply to: [66284121](https://github.com/Microsoft/ChakraCore/pull/1096#discussion_r66284121) [](ancestors = 66284121)